packages/SettingsLib/src/com/android/settingslib/bluetooth/BluetoothTimeoutReceiver.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2020 The Calyx Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.settingslib.bluetooth;
+
+import android.app.AlarmManager;
+import android.app.PendingIntent;
+import android.bluetooth.BluetoothAdapter;
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.util.Log;
+
+import java.util.Date;
+
+public class BluetoothTimeoutReceiver extends BroadcastReceiver {
+    private static final String TAG = "BluetoothTimeoutReceiver";
+
+    private static final String INTENT_TIMEOUT = "android.bluetooth.intent.TIMEOUT";
```

Comment:
Might be good to make this a protected broadcast so that only the system can send this broadcast. You'd have to add this String to the AndroidManifest.xml here:

https://github.com/GrapheneOS/platform_frameworks_base/blob/0697a2cd6eca819645db88586ecc85b8d11a1c32/core/res/AndroidManifest.xml#L25

Reply:
Sory, im really new in this, you mean to add this string:
<protected-broadcast android:name="android.bluetooth.intent.TIMEOUT" /> in AndroidManifest.xml?
Do i have to change something in this java file as well?

Reply:
Yeah, just add it to the manifest, don't need to do anything with the Java code. [(The OS will check and enforce that only system code sends protected-broadcasts.)](https://github.com/GrapheneOS/platform_frameworks_base/blob/11/services/core/java/com/android/server/am/ActivityManagerService.java#L16016-L16050)

Could probably put it at the end of the protected-broadcast section to make porting between Android versions easier:

https://github.com/GrapheneOS/platform_frameworks_base/blob/0697a2cd6eca819645db88586ecc85b8d11a1c32/core/res/AndroidManifest.xml#L656-L661

Reply:
Also, I noticed that this implementation is from CalyxOS. If this protected-broadcast addition isn't already in CalyxOS, you could also submit this change to them.

Reply:
Done. Going to register at gitlab and submit to Calyx.
Thanks